pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -211,7 +213,11 @@ def test_stringified_slice_with_tz(self):
         start = datetime.datetime.now()
         idx = DatetimeIndex(start=start, freq="1d", periods=10)
         df = DataFrame(lrange(10), index=idx)
-        df["2013-01-14 23:44:34.437768-05:00":]  # no exception here
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
use ``tm.assert_raises_warning``

Reply:
I pulled the latest changes from master but i could not find a `tm.assert_raises_warning`, there is only a `tm.assert_raises_regex` which is meant for errors, not warnings and a  `assert_produces_warning` which is most suitable for this situation. I will take that one.